spec/features/checkout_spec.rb

```diff
@@ -147,8 +147,14 @@
 
   def fill_in_address
     address = "order_bill_address_attributes"
-    fill_in "#{address}_firstname", with: "Ryan"
-    fill_in "#{address}_lastname", with: "Bigg"
+
+    if ::Spree.solidus_gem_version < Gem::Version.new('2.11.x')
+      fill_in "#{address}_firstname", with: "Ryan"
+      fill_in "#{address}_lastname", with: "Bigg"
+    else
+      fill_in "#{address}_name", with: "Ryan Bigg"
+    end
+
```

Comment:
I think it would be better to use the new Solidus preference to make this choice:

```ruby
    def fill_in_address
      address = "order_bill_address_attributes"
      if use_address_full_name?
        fill_in "#{address}_name", with: "Ryan Bigg"
      else
        fill_in "#{address}_firstname", with: "Ryan"
        fill_in "#{address}_lastname", with: "Bigg"
      end
      fill_in "#{address}_address1", with: "143 Swan Street"
      fill_in "#{address}_city", with: "Richmond"
      select "Ohio", from: "#{address}_state_id"
      fill_in "#{address}_zipcode", with: "12345"
      fill_in "#{address}_phone", with: "(555) 555-5555"
    end

    def use_address_full_name?
      Spree::Config.has_preference?(:use_combined_first_and_last_name_in_address) &&
        Spree::Config.use_combined_first_and_last_name_in_address
    end
```
WDYT?

Reply:
😱 I did not know that this was configurable. does that means that shops will be able to keep the "old way" running with a first name and a lastname after upgrading to 2.11?

I'll update the code to use this config flag

Reply:
Thank you!
Yes, in 2.11 there will be a deprecation warning and I think that the "old way" way will be removed in 3.0 👍